web/src/navigator.rs

```diff
@@ -60,39 +44,33 @@ impl NavigatorBackend for WebNavigatorBackend {
         window_spec: Option<String>,
         vars_method: Option<(NavigationMethod, IndexMap<String, String>)>,
     ) {
-        const JAVASCRIPT_PREFIX: &str = "javascript:";
+        // If the URL is empty, we ignore the request
+        if url.is_empty() {
+            return;
+        }
 
         if let Some(window) = window() {
-            let url_trimmed = url.trim();
-            if url_trimmed
-                .get(..JAVASCRIPT_PREFIX.len())
-                .unwrap_or_default()
-                .eq_ignore_ascii_case(JAVASCRIPT_PREFIX)
-            {
-                let target = window_spec.unwrap_or_else(|| "".to_string());
-                if target.is_empty() || target == "_self" || target == "undefined" {
-                    self.run_script(&url);
+            let url = match Url::parse(&url) {
```

Comment:
`url_from_relative_url` can be used here.

Reply:
Doing so would regress #2214, I think?

Reply:
Did you mean to reply for my comment about `pre_process_url`? Because this comment is a pure code deduplication.